.github/workflows/security.yml

```
@@ -1,49 +1,18 @@
 name: SecurityChecks
 on:
+  workflow_dispatch:
   pull_request: {}
   push:
     branches: ["master"]
 jobs:
-  semgrep:
-    name: Scan
-    runs-on: ubuntu-latest
-    container:
-      image: returntocorp/semgrep
-    steps:
-      - uses: actions/checkout@v2
-      - run: semgrep ci
-        env:
-          SEMGREP_APP_TOKEN: ${{ secrets.SEMGREP_APP_TOKEN }}
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  security-sast:
+    uses: razorpay/security-action/.github/workflows/semgrep.yml@master
+    secrets:
+      SEMGREP_APP_TOKEN: ${{ secrets.SEMGREP_APP_TOKEN }}
 
-  workflow_status:
-    runs-on: ubuntu-latest
-    name: Update Status Check
-    needs: [ semgrep ]
+  security-statuscheck:
+    needs: [ security-sast ]
     if: always()
-    env:
-      githubCommit: ${{ github.event.pull_request.head.sha }}
-    steps:
-      - name: Set github commit id
-        run: |
-          if [ "${{ github.event_name }}" = "push" ]; then
-            echo "githubCommit=${{ github.sha }}" >> $GITHUB_ENV
-          fi
-          exit 0
-      - name: Failed
-        id: failed
-        if: contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled')
-        run: |
-          echo 'Failing the workflow for github security status check.'
-          curl -X POST -H "Content-Type: application/json" -H "Authorization: token ${{ github.token }}" \
-          -d '{ "state" : "failure" , "context" : "github/security-status-check" , "description" : "github/security-status-check", "target_url" : "https://github.com/${{ github.repository }}" }' \
-          https://api.github.com/repos/${{ github.repository }}/statuses/${{ env.githubCommit }}
-          exit 1
-      - name: Success
-        if: steps.failed.conclusion == 'skipped'
-        run: |
-          echo 'Status check has passed!'
-          curl -X POST -H "Content-Type: application/json" -H "Authorization: token ${{ github.token }}" \
-          -d '{ "state" : "success" , "context" : "github/security-status-check" , "description" : "github/security-status-check", "target_url" : "https://github.com/${{ github.repository }}" }' \
-          https://api.github.com/repos/${{ github.repository }}/statuses/${{ env.githubCommit }}
-          exit 0
+    uses: razorpay/security-action/.github/workflows/status_check.yml@master
+    with:
+      WORKFLOW_RESULT: ${{ needs.security-sast.result == 'success' && needs.security-sast.result != 'cancelled' || 'false' }}
```

I don't think you can use a private workflow file ( for github action runner, it is private ) directly without passing any PAT tokens.

Try something like this:

```suggestion
    - uses: actions/checkout@v3
      with:
        repository: razorpay/security-action
        ref: master
        token: ${{ secrets.GitHub_PAT }} # `GitHub_PAT` is a secret that contains your PAT to clone that private repo
      
    - name: Run the private security status_check action
      uses: .github/workflows/status_check.yml
      with:
         WORKFLOW_RESULT: ${{ needs.security-sast.result == 'success' && needs.security-sast.result != 'cancelled' || 'false' }}
```